### DISCUSSION TITLE
Integration Tests and configuration

### DISCUSSION THREAD

**Discussion by enrico23** [microsoft/aspire#4843] [General]
Hi everyone 🙂,
I am trying to write integration with Aspire and following the article on https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/testing.

The `DistributedApplicationTestingBuilder` wraps the `DistributedApplicationBuilder` of the Aspire app host. This allows to spin up the orchestrator before running the tests. 

I slightly changed the configuration of the orchestrator to use secrets for some of the required credentials. I don't want to have them in the code and visible on my GitHub repositories.  

The plan is to use the .NET Secret Manager to inject them for my development environment and Azure KeyVault or any other type of server ingestion mechanism in the production environments.

The `DistributedApplicationTestingBuilder` doesn't allow to modify the configuration.

The simple solution is for the test environment to inject the same environment variables.

I was wondering if there is a way to modify the configuration before building the host: 
```
 var appHost = await DistributedApplicationTestingBuilder.CreateAsync<Projects.MyProjectReference>();
 // Modify configuration
await using var app = await appHost.BuildAsync();
 ```
 
Is there a way for the test runner to launch a different profile for the Aspire app host? 
 
 Thanks in advance 🙂

**Comment by enrico23** [comment]
This project helped me to undertstand some of the internals and solve my issue
https://github.com/dotnet/aspire-samples/tree/main/tests

**Reply by jflaga** [reply]
Hi @enrico23 , were you able to inject environment variables from your test environment?

**Reply by weitzhandler** [reply]
It's possible to change the configuration via `testingBuilder.Configuration.Add` but it's not picked up by the individual resources.
